src/algorithms/calorimetry/CalorimeterHitDigi.cc

```diff
@@ -142,6 +150,12 @@ void CalorimeterHitDigi::process(
         ix++;
     }
 
+    // create hit and association in advance
+    edm4hep::MutableRawCalorimeterHit rawhit;
+#if EDM4EIC_VERSION_MAJOR >= 7
+    std::vector<edm4eic::MutableMCRecoCalorimeterHitAssociation> rawassocs_;
+#endif
```

Comment:
Note to reviewer: we change from a `collection->create()` at the end to creating mutables up front because we do not know in advance whether a raw hit will be produced (it could fall out of time), yet we don't want to repeat the loop over sim hits to be merged after we have decided that we did produce a hit. Avoiding the second loop to add associations means we need to have a raw hit to associate with, which means we need to create the mutable in advance.